packages/store-engine-bro-fs/src/index.ts

```diff
@@ -0,0 +1,210 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {CRUDEngine} from '@wireapp/store-engine';
+import {
+  RecordAlreadyExistsError,
+  RecordNotFoundError,
+  RecordTypeError,
+  UnsupportedError,
+} from '@wireapp/store-engine/dist/commonjs/engine/error/';
+import * as fs from 'bro-fs';
+
+export interface FileSystemEngineOptions {
+  fileExtension: string;
+  size: number;
+  type: number;
+}
+
+const TEN_MEGABYTES = 1024 * 1024 * 10;
+
+export class FileSystemEngine implements CRUDEngine {
+  public storeName = '';
+
+  private config: FileSystemEngineOptions = {
+    fileExtension: '.dat',
+    size: TEN_MEGABYTES,
+    type: typeof window === 'undefined' ? 0 : window.TEMPORARY,
+  };
+
+  constructor() {}
+
+  public async isSupported(): Promise<void> {
+    if (typeof window === 'undefined' || !fs.isSupported()) {
+      const message = `File and Directory Entries API is not available on your platform.`;
```

Comment:
"File and Directory Entries API"?

Reply:
That's what it is called: https://developer.mozilla.org/en-US/docs/Web/API/File_and_Directory_Entries_API

Reply:
Ah okay :)